Feature/add tenants for get companies on prem

JWK opened the pull request:
Enhanced the GetAllCompaniesOfEnvironment procedure to differentiate between SaaS and on-premises environments.
Updated the base URL construction for on-premises environments to include the tenant name as a query parameter.
Ensured that the HTTP request method remains consistent for both environment types.

JWK committed:
This update improves the flexibility of the GetAllCompaniesOfEnvironment procedure in Communication Codeunit, allowing it to function correctly in both SaaS and on-premises environments.

JWK committed:
Update GetAllCompaniesOfEnvironment to Handle On-Premises Environments

This update improves the flexibility of the GetAllCompaniesOfEnvironment procedure, allowing it to function correctly in both SaaS and on-premises environments.

JWK committed:
This update improves the flexibility of the GetAllCompaniesOfEnvironment procedure, allowing it to function correctly in both SaaS and on-premises environments.

JWK committed:
Update Communication.Codeunit.al

Added labels instead of text

JWK committed:
Clean if else statement

Removed begin ends

Bertverbeek4PS commented:
@JWK thanks for your PR.
Could you also change the tooltip on page 50102 "ENVHUB Setup"
Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc/api/v2.0/v1.0/companies.

to

Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc

Bertverbeek4PS commented on src/Communication.Codeunit.al:
```diff
@@ -45,12 +45,14 @@ codeunit 50103 "ENVHUB Communication"
+        if EnvironmentInformation.IsSaaS() then begin
+            exit(ENVHUBHttp.RequestMessage(StrSubstNo(saasUrlLbl, ENVHUBSetup."Tenant ID", environmentName), Method::Get, ''))
+        end else begin
+            exit(ENVHUBHttp.RequestMessage(StrSubstNo(onPremUrlLbl, ENVHUBSetup."Base URL", environmentName), Method::Get, '')); //Change of baseUrl with tenant/environment name
```
Please remove the comment

Bertverbeek4PS replied:
@JWK I meant the comment on line 55.
In the labels there must always be a comment.

JWK replied:
Removed comment on line 55 and added the label comments.

JWK committed:
Update Setup.Page.al

Update tooltip for OnPrem BaseUrl

JWK commented:
> @JWK thanks for your PR. Could you also change the tooltip on page 50102 "ENVHUB Setup" Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc/api/v2.0/v1.0/companies.
> 
> to
> 
> Specifies the redirect base URL api of the application that will be used to the Business Central integration. Like https://nav.contoso.com:7048/bc

I’ve updated the tooltip.

Bertverbeek4PS commented on src/Communication.Codeunit.al:
```diff
@@ -45,12 +45,13 @@ codeunit 50103 "ENVHUB Communication"
         EnvironmentInformation: Codeunit "Environment Information";
         Method: Option Get,Post,Patch;
-        UrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies', comment = '%1 = Tenant ID, %2 = Environment name';
+        saasUrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies';
```
Please add the comment here again

JWK replied:
Added the following comment for SaasUrlLbl  '%1 = Tenant ID, %2 = Environment name';

Bertverbeek4PS commented on src/Communication.Codeunit.al:
```diff
@@ -45,12 +45,13 @@ codeunit 50103 "ENVHUB Communication"
         Method: Option Get,Post,Patch;
-        UrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies', comment = '%1 = Tenant ID, %2 = Environment name';
+        saasUrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies';
+        onPremUrlLbl: Label '%1/api/v1.0/companies?tenant=%2';
```
Please add the comment here again

JWK replied:
Added the following comment for OnPremUrlLbl '%1 = Base URL, %2 = Environment name';

JWK committed:
Updated comments for SaasUrlLbl and OnPremUrlLbl labels with comment in Communication.Codeunit.al.
SaasUrlLbl: Label 'https://api.businesscentral.dynamics.com/v2.0/%1/%2/api/v1.0/companies', comment = '%1 = Tenant ID, %2 = Environment name';
OnPremUrlLbl: Label '%1/api/v1.0/companies?tenant=%2', comment = '%1 = Base URL, %2 = Environment name';
Removed the unnecessary comment on line 55.
